File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math 
from collections import deque

def solution(progresses, speeds):
    answer = []
    remain = deque()
    
    cnt = 1
    
    for i in range(len(progresses)):
        x  = math.ceil((100 - progresses[i])/speeds[i])
        remain.append(x)
        
    
    while remain:
        first_task = remain.popleft()
        cnt = 1
        
        while remain and remain[0] <= first_task:
            remain.popleft()
            cnt += 1
            
        answer.append(cnt)
    
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
     const days = progresses.map((p, i) => Math.ceil((100 - p) / speeds[i]));
    
    while(days.length > 0){
        let first_task = days.shift();
        let cnt = 1;
        
        while(days.length > 0 && days[0] <= first_task){
            days.shift();
            cnt += 1
        }
        
        answer.push(cnt)
    }

    
    console.log(answer)
    
    
    return answer;
}